specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2022-10/schemaregistry.json

```
@@ -0,0 +1,614 @@
+              "type": "string",
+              "format": "binary"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content",
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema group, schema name, and version."
+              },
+              "Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "Schema-Id-Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema ID."
+              },
+              "Schema-Group-Name": {
+                "type": "string",
+                "description": "References schema group."
+              },
+              "Schema-Name": {
+                "type": "string",
+                "description": "References schema name."
+              },
+              "Schema-Version": {
+                "type": "integer",
+                "description": "Version of the returned schema."
+              }
+            }
+          },
+          "415": {
+            "description": "Unsupported Media Type - Content type of the supplied schema is unsupported by schema group.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code for specific error that occurred."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          },
+          "default": {
+            "description": "An error response received from the Azure Schema Registry service.",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "Error code for specific error that occurred."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/$schemaGroups/{groupName}/schemas/{schemaName}": {
+      "put": {
+        "summary": "Register new schema",
+        "description": "Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.\n",
+        "x-ms-examples": {
+          "Register schema": {
+            "$ref": "./examples/OperationSchema_Register.json"
+          }
+        },
+        "operationId": "Schema_Register",
+        "consumes": [
+          "application/json; serialization=Avro",
+          "application/json",
+          "application/json; serialization=Json",
```

Should it also accept `schema+json`?
```suggestion
          "application/json; serialization=Json",
          "application/schema+json",
```

---

I would vote to standardize on one content type, but perhaps we can raise this question with the REST board in the email thread.

---

I'm going to add it for now to reflect everything that the service can support then share in the email thread.

---

After some follow up, for the request and response headers for Json, we'll follow the same pattern as Avro: "application/json; serialization=Json"

---

Should it just be:
`application/json; serialization=json`
i.e. not capitalize json.

---

The service treats headers as case insensitive (which is why we kept "Avro" capitalized). Is this an SDK requirement? If so, I can push an update to reflect that.

---

No it shouldn't impact the SDK, but I mentioned this because I've never seen JSON capitalized that way - in the content-type `application/json` it is all lower-case.